integration_tests/src/main/python/fastparquet_compatibility_test.py

```diff
@@ -40,14 +39,15 @@ def read_with_fastparquet_or_plugin(spark):
         if plugin_enabled:
             return spark.read.parquet(data_path)
         else:
-            import fastparquet
             df = fastparquet.ParquetFile(data_path).to_pandas()
             return spark.createDataFrame(df)
 
     return read_with_fastparquet_or_plugin
 
 
-@pytest.mark.parametrize('corrected_conf', [rebase_write_corrected_conf])
+@pytest.mark.skipif(condition=spark_version() < "3.4.1",
+                    reason="spark_df.to_pandas() is not reliable on prior versions of Spark. 3.4.1 and above seem to "
```

Comment:
nit: Is that really the problem? We are going from fastparquet->pandas->spark df, so I am not sure spark_df.to_pandas is correct.

Could you also add a comment somewhere explaining a bit more details about what the failures are. It could just be a response here. I just want to be sure that we either file issues with fastparquet, spark, etc to get them fixed or we know that they are fixed in later versions already.

Reply:
> Is that really the problem? 

No, this is mislabeled. This should have mentioned `sparkSession.createDataFrame(pandasDf)` instead. I'll fix that shortly.

> explaining a bit more details about what the failures are...

I'll put those in a separate comment on this bug, and group them.